### ISSUE TITLE
read_csv falsely detect NA in integer column

### ISSUE THREAD

**Issue by jangorecki** [dask/dask#1532]
It seems that NA's are mistakenly detected in my second integer column, thus whole column is read as float64.
When I load data I get my integer column converted to float64, specifying dtypes will raise exception on NAs in integer. The problem is there are no NA in data.
```py
x = dd.read_csv("hdfs://path/X1e7_2c.csv")
x.dtypes
#KEY      int64
#X2     float64
#dtype: object
x[x.KEY.isnull()].compute()
#Empty DataFrame
#Columns: [KEY, X2]
#Index: []
x[x.X2.isnull()].compute()
#Empty DataFrame
#Columns: [KEY, X2]
#Index: []
```

I manage to workaround this using the following, let me know if it can be improved/simplified.
```py
x = dd.read_csv("hdfs://path/X1e7_2c.csv", na_filter=False)
x.dtypes
#KEY     int64
#X2     object
#dtype: object
x[x.KEY.isnull()].compute()
#Empty DataFrame
#Columns: [KEY, X2]
#Index: []
x[x.X2.isnull()].compute()
#Empty DataFrame
#Columns: [KEY, X2]
#Index: []
x[['X2']] = x[['X2']].astype(int)
x.dtypes
#KEY    int64
#X2     int64
#dtype: object
```

**Comment by mrocklin** [comment]
Does the CSV file fit in pandas?  If so what does it say are the dtypes?

```python
import hdfs3
hdfs = hdfs3.HDFileSystem()

with hdfs.open("/path/X1e7_2c.csv") as f:
    df = pd.read_hdf(f)

print(df.dtypes)
```

**Comment by jangorecki** [comment]
I'm getting the following error from on `pd.read_hdf`.
```py
#.../pandas/io/pytables.py", line 320, in read_hdf
#    raise NotImplementedError('Support for generic buffers has not been '
#NotImplementedError: Support for generic buffers has not been implemented.
```
`hdfs.ls` lists files well.

I have the same csv locally, and here pandas reads properly.
```py
xx = pd.read_csv("X1e7_2c.csv")
xx.dtypes
#KEY    int64
#X2     int64
#dtype: object
```

**Comment by mrocklin** [comment]
Ah, sorry, I was dumb and listed read_hdf rather than read_csv

My main question was really "How does Pandas handle this data?  Does it make the same mistake?"

**Comment by jangorecki** [comment]
Looks good with `read_csv`
```py
#df.dtypes               
#KEY    int64
#X2     int64
#dtype: object
```

**Comment by jcrist** [comment]
I suspect this is the same issue that was fixed in https://github.com/dask/dask/pull/1495. Can you try installing from git and see if that fixes things?

`pip install -U git+https://github.com/dask/dask.git`

**Comment by jangorecki** [comment]
@jcorbin just tried and got the same behaviour, but not sure if actual update took place. Should I also upgrade it on nodes? Is there any way to get git commit hash of currently running dask?

**Comment by jcrist** [comment]
The fix should only matter locally (on the head node), but generally it's a good idea to have the same versions of things running everywhere. Note that the dask code that runs read_csv distributed is the same that runs it locally, so you can test on your machine using a local csv as well.

If you're sure you ran it on master and got the same issue, then I'm not sure what's going on. I'd need a reproducible example to help debug.

**Comment by jangorecki** [comment]
I have another 1e7 dataset from same random number generator and this one loads well as int64, so this heavily depends on the data, will provide dataset soon, via email.

**Comment by mrocklin** [comment]
If you can please try to keep things public.

**Comment by jangorecki** [comment]
@mrocklin sure [mega.nz X1e7_2c.tar.gz](https://mega.nz/#!JpgXHCTA!fIntjGl-XHcea-xc_6P8idt_MkylnOEczTxSs7fRSdM)

**Comment by jcrist** [comment]
Apologies for the delay, this has been fixed in #1571, and confirmed to work on the provided file:

```python
In [1]: import dask.dataframe as dd

In [2]: df = dd.read_csv('test.csv')

In [3]: df.dtypes
Out[3]:
KEY    int64
X2     int64
dtype: object
```
